### PR TITLE
Kernel: get_disk renamed to get_disk_and_module

### PR DESCRIPTION
**Pull request by dinatale2**
### Description

As of https://github.com/torvalds/linux/commit/fb6d47a, `get_disk`
is now `get_disk_and_module`. Add a configure check to determine
if we need to use `get_disk_and_module`.

### Motivation and Context

Keep up with upstream kernel.

### How Has This Been Tested?

Kernel Builders will test if this builds.

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [ ] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (a change to man pages or other documentation)

### Checklist:

- [x] My code follows the ZFS on Linux code style requirements.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [x] All commit messages are properly formatted and contain `Signed-off-by`.
- [ ] Change has been approved by a ZFS on Linux member.

### CODE REVIEW

File: config/kernel-get-disk-and-module.m4 [hunk 1]
```diff
@@ -0,0 +1,19 @@
+dnl #
+dnl # 4.16 API change
```
Comment (loli10K) [inline config/kernel-get-disk-and-module.m4]: nit: i think this change landed in linux-block/for-4.17

File: config/kernel-get-disk-and-module.m4 [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+dnl # Verify if get_disk_and_module() symbol is available.
+dnl #
+AC_DEFUN([ZFS_AC_KERNEL_GET_DISK_AND_MODULE],
+	[AC_MSG_CHECKING([whether get_disk_and_module() is available])
+	ZFS_LINUX_TRY_COMPILE_SYMBOL([
+		#include <linux/genhd.h>
+	], [
+		struct gendisk *disk = NULL;
+                (void) get_disk_and_module(disk);
```
Comment (loli10K) [inline config/kernel-get-disk-and-module.m4]: nit: replace spaces with hard tab

File: config/kernel-get-disk-and-module.m4 (hunk 1 above)
Comment (dinatale2) [reply config/kernel-get-disk-and-module.m4]: I was looking at https://github.com/torvalds/linux and they are currently in `4.16-rc` releases. Am I looking at the wrong place for version numbers?

Comment (richardelling) [reply config/kernel-get-disk-and-module.m4]: https://stackoverflow.com/questions/30268332/why-does-the-linux-kernel-repository-have-only-one-branch

Comment (loli10K) [reply config/kernel-get-disk-and-module.m4]: @dinatale2 i found the change here http://git.kernel.dk/cgit/linux-block/log/?h=for-4.17/block (http://git.kernel.dk/cgit/linux-block/commit/?h=for-4.17/block&id=3079c22ea815775837a4f389ce2f7e1e7b202e09) and thought this was going to land in 4.17. It's just a nit though, the functional aspect is the important one and LGTM.

File: module/zfs/zvol.c [hunk 3]
```diff
@@ -1589,7 +1589,11 @@ zvol_probe(dev_t dev, int *part, void *arg)
 	struct kobject *kobj;
 
 	zv = zvol_find_by_dev(dev);
+#ifndef HAVE_GET_DISK_AND_MODULE
 	kobj = zv ? get_disk(zv->zv_disk) : NULL;
+#else
+	kobj = zv ? get_disk_and_module(zv->zv_disk) : NULL;
+#endif
```
Comment (behlendorf) [inline module/zfs/zvol.c]: nit: let's move this in to `include/linux/blkdev_compat.h`, maybe something alone the lines of `get_disk_ro`.

```c
#ifndef HAVE_GET_DISK_AND_MODULE
static inline struct kobject *
get_disk_and_module(struct gendisk *disk)
{
        return (get_disk(disk));
}
#endif
```

**Pull request merged**